onboarding/source/amb_light_service.c

```diff
@@ -33,21 +47,62 @@ static void lightServiceTask(void * pvParameters);
 
 obc_error_code_t initLightService(void) {
     /* USER CODE BEGIN */
-    // Create the task and queue here. Return error code if task/queue was not created successfully.
+    BaseType_t xReturned = pdFAIL;
+        if (lightServiceHandle == NULL) {
+            xReturned = xTaskCreate(lightServiceTask,
+                                    LIGHT_SERVICE_NAME,
+                                    LIGHT_SERVICE_STACK_SIZE,
+                                    NULL,
+                                    LIGHT_SERVICE_PRIORITY,
+                                    &lightServiceHandle);
+                                     }
+        if (xReturned == pdFAIL) {
+             return OBC_ERR_CODE_TASK_CREATION_FAILED;
+        
+        }
+
+        if (lightserviceQueue == NULL) {
+                lightserviceQueue = xQueueCreate(LIGHT_SERVICE_SIZE, LIGHT_QUEUE_SIZE);
 
+                if (lightserviceQueue == NULL) {
+                     return OBC_ERR_CODE_QUEUE_FULL;
+                }
+        }
+    return 1;                            
     /* USER CODE END */
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    light_event_t eventReceived;
+
+
+ 	while (1) {
+        if (xQueueReceive(lightserviceQueue, &eventReceived, portMAX_DELAY) == pdPASS) {
+            adcData_t adcData;
+            if (eventReceived == MEASURE_LIGHT) {
+                adcStartConversion(adcREG1, adcGROUP1);
+                while (!adcIsConversionComplete(adcREG1, adcGROUP1));
+                adcGetData(adcREG1, adcGROUP1, &adcData);
+
+                char str[32];
+                    snprintf(str, 32, "The ambient light is %d", adcData.value);
```

Comment:
`adcData.value` is a `uint16_t` if I remember correctly. Use the `%u` format specifier instead